hub/core/dataset/dataset.py

```diff
@@ -387,6 +392,12 @@ def create_tensor(
         if info_kwargs:
             tensor.info.update(info_kwargs)
         self.storage.maybe_flush()
+        if create_id_tensor:
+            id_tensor_name = get_sample_id_tensor_key(name)
+            self.create_tensor(id_tensor_name, hidden=True, create_id_tensor=False)
+            self._link_tensors(
```

Comment:
(later) this newly tensor should have properties of Merkle trees (which would be tricky to treat them as tensors, but maybe a function of top of tensor of hashes).

Just saying to keep architecture intact latre.